### PR TITLE
[14.0] [FIX] sale_quick: multi company

### PR DESCRIPTION
**Pull request by renda-dev**
Simple PR to set a domain in the "quick add" view to filter out products that do not belong to the SO's company and products not for sale.

### CODE REVIEW

**Review by francesco-ooops — APPROVED** [review]
Functional ok!

@renda-dev let's mark it as a FIX

**Review by legalsylvain — CHANGES_REQUESTED** [review]
I dont think it's a generic change.
Some people could want to sale product from other company, or without company.

I think you should change a rule in your instance.

**Comment by francesco-ooops** [comment]
Hi @legalsylvain , I think we're just aligning behavior of sale_quick to standard odoo sale order line domain, as you can't add a product belonging exclusively to Company 1 to a sale order belonging to Company 2. 

Indeed, this is the error we're fixing with current version:

image

Products with no company assigned can still be added to any sale order line.

**Comment by legalsylvain** [comment]
Oh ! I see ! So you're right, but i propose a différent implémentation : 
- get the domain defined in the field product_id https://github.com/odoo/odoo/blob/14.0/addons/sale/models/sale.py#L1396
- return it.

That way : 
- it avoid to duplicate code
- if domain of sale.order.line. product_id changes, the domain in sale quick will change automatically.

**Review by legalsylvain — APPROVED** [review]
Thanks ! 

CC : @bealdav.

**Comment by francesco-ooops** [comment]
@pedrobaeza merge please?

File: sale_quick/models/sale_order.py
Comment (pedrobaeza) [inline sale_quick/models/sale_order.py:31]: This is missing possible fields values, as domains may be "dynamic". Example: `"[('type_id', '=', type_id)]"`

**Review by pedrobaeza — DISMISSED** [review]

File: sale_quick/models/sale_order.py
Comment (renda-dev) [reply sale_quick/models/sale_order.py:31]: Thanks for the feedback, however, I'm not sure how to go about fixing it... What would you suggest?

Comment (legalsylvain) [reply sale_quick/models/sale_order.py:31]: > This is missing possible fields values, as domains may be "dynamic". Example: "[('type_id', '=', type_id)]"

You're right.
however, I think that this case is quite theoritical and will not occures. and if it occures, it should be fixed by a custom glue module.
I mean, if a custom module add a domain like "[('type_id', '=', type_id)]" and if sale_quick is installed, the custom module (or another glue module) should update the function. _get_domain_add_products.
(with or without this patch).

For me, this implementation is OK.

I see it also robust because if the domain changes in the sale module, we have nothing to do in the OCA module.

Comment (pedrobaeza) [reply sale_quick/models/sale_order.py:31]: OK, I dismiss my request changes if you want to go ahead, but I don't think this solution is good enough. Maybe this can be handled at `fields_view_get` level.

Comment (francesco-ooops) [reply sale_quick/models/sale_order.py:31]: @pedrobaeza would you like for us to add a line in roadmap?

Comment (pedrobaeza) [reply sale_quick/models/sale_order.py:31]: As you prefer. I just refrain for this.

**Comment by francesco-ooops** [comment]
@OCA/crm-sales-marketing-maintainers anyone available for merge? 

Or if you have ideas about another implementation, we're open to find a solution that makes everyone happy

**Comment by francesco-ooops** [comment]
@damdam-s as module creator, do you have any input on how to implement this in the best way? thank you!

**Comment by francesco-ooops** [comment]
@yajo do you have any suggestion on how to approach this the best way?

**Comment by francesco-ooops** [comment]
@simahawk do you have any suggestions on how to get out of this deadlock? appreciate it ;)

**Comment by francesco-ooops** [comment]
@pedrobaeza seems your comment was blocking indeed 😓

**Comment by francesco-ooops** [comment]
@legalsylvain can you help us get unstuck? appreciate it

**Comment by legalsylvain** [comment]
Hi. i'm not a PSC of this repo. I have no merge right. should wait for PSC / maintainer of the module.
regards.

**Comment by francesco-ooops** [comment]
@legalsylvain thank you, I was not asking for a merge but rather on whether you could think of a solution that would please @pedrobaeza since you suggested this approach in the early steps of implementation 

unfortunately I can't seem to get @OCA/crm-sales-marketing-maintainers to merge since a month even if Pedro has marked his comment as not blocking :/

**Comment by francesco-ooops** [comment]
@pedrobaeza can you merge this anyway please?

**Comment by pedrobaeza** [comment]
OK, I think a crash will appear while evaluing the domain that way, but let's see:

/ocabot merge patch

**Pull request merged**
